src/web/app/pages/PromarkSettings/index.tsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { Flex } from 'antd';
+import classNames from 'classnames';
+
+import useI18n from 'helpers/useI18n';
+import deviceMaster from 'helpers/device-master';
+import applyRedDot from 'helpers/device/promark/apply-red-dot';
+import {
+  defaultField,
+  defaultGalvoParameters,
+  defaultRedLight,
+} from 'app/constants/promark-constants';
+import storage from 'implementations/storage';
+import FieldBlock from 'app/components/dialogs/promark/FieldBlock';
+import RedDotBlock from 'app/components/dialogs/promark/RedDotBlock';
+import LensBlock from 'app/components/dialogs/promark/LensBlock';
+import dialogCaller from 'app/actions/dialog-caller';
+import promarkDataStore from 'helpers/device/promark/promark-data-store';
+import { getWorkarea } from 'app/constants/workarea-constants';
+
+import styles from './index.module.scss';
+
+export default function PromarkSettings(): JSX.Element {
+  const { initialize: t } = useI18n();
+  const [field, setField] = useState(defaultField);
+  const [redDot, setRedDot] = useState(defaultRedLight);
+  const [galvoParameters, setGalvoParameters] = useState(defaultGalvoParameters);
+
+  const isInch = useMemo(() => storage.get('default-units') === 'inches', []);
+  const serial: string = useMemo(() => storage.get('last-promark-serial'), []);
+  const { width } = useMemo(() => getWorkarea('fpm1') ?? { width: 110 }, []);
```

Comment:
Optional: Not sure if this `??` part needed because `getWorkarea` would use default value